File: microservice/microservice-app/src/main/java/com/greg/microservice/data/CustomerEntity.java
package com.greg.microservice.data;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Data
public class CustomerEntity {
    @Column( columnDefinition = "TIMESTAMP",nullable = false, updatable = false)
    @CreationTimestamp
    protected LocalDateTime created;

    @Id
    @GeneratedValue
    protected UUID id;

    private String name;
}

File: microservice/microservice-app-kotlin/src/main/kotlin/com/greg/microserviceappkotlin/data/CustomerEntity.kt
package com.greg.microserviceappkotlin.data

import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
class CustomerEntity (
    @Column(columnDefinition = "TIMESTAMP", nullable = false, updatable = false)
    @CreationTimestamp
    val created: LocalDateTime,

    @Id
    @GeneratedValue
    val id: UUID,
    val name: String
)